lib/utils/fs_test.go

```diff
@@ -38,7 +40,32 @@ func TestOpenFileLinks(t *testing.T) {
 	// broken-s -> nonexistent
 	// hardfile
 	// hardfile-h -> hardfile
-	rootDir := t.TempDir()
+
+	// macOS is special cased here since t.TempDir() returns a path to a protected directory that doesn't allow symlinks.
+	var rootDir string
+	switch runtime.GOOS {
+	case "darwin":
+		suffix := make([]byte, 8)
+		_, err := rand.Read(suffix)
+		if err != nil {
+			t.Fatalf("failed to generate random suffix: %v", err)
```

Comment:
Should we use `require.NoError` for consistency with all of our other tests?

Reply:
fixed

Reply:
Where? I'm not seeing the fix?